src/components/autocomplete/js/autocompleteController.js

```diff
@@ -635,10 +635,11 @@ function MdAutocompleteCtrl ($scope, $element, $mdUtil, $mdConstant, $mdTheming,
   function fetchResults (searchText) {
     var items = $scope.$parent.$eval(itemExpr),
         term  = searchText.toLowerCase(),
-        isList = angular.isArray(items);
+        isList = angular.isArray(items),
+        isPromise = !!items.then; // Every promise should contain a `then` property
 
-    if ( isList ) handleResults(items);
-    else          handleAsyncResults(items);
+    if (isList) handleResults(items);
+    else if (isPromise) handleAsyncResults(items);
```

Comment:
I thought about doing nothing, because it makes no sense, and otherwise it will prevent the autocomplete from working. See the demo of the actual issue. http://codepen.io/anon/pen/Gowjmv?editors=1010